src/connect.rs

```diff
@@ -72,15 +72,19 @@ pub fn connect_to(peer_info: CrustInfo, send_after_connect: Option<WireMsg>) ->
 
 fn handle_new_connection_res(
     peer_addr: SocketAddr,
-    new_peer_conn_res: Result<quinn::NewClientConnection, quinn::ConnectionError>,
+    new_peer_conn_res: Result<quinn::NewConnection, quinn::ConnectionError>,
 ) {
     let new_peer_conn = match new_peer_conn_res {
         Ok(new_peer_conn) => new_peer_conn,
         Err(e) => return handle_connect_err(peer_addr, &From::from(e)),
     };
+    current_thread::spawn(new_peer_conn.driver.map_err(move |e| {
+        communicate::handle_communication_err(peer_addr, &From::from(e), "Connection driver failed")
```

Comment:
this should call `handle_connect_err` as that is specialised to deal with connect errors